snapcraft/parts/lifecycle.py

```diff
@@ -90,6 +92,8 @@ def run(command_name: str, parsed_args: "argparse.Namespace") -> None:
 
     build_count = utils.get_parallel_build_count()
 
+    partitions = _validate_and_get_partitions(yaml_data)
```

Comment:
Try unmarshalling just the components keyword (and base, type, build-base possibly)

Reply:
allow for core22 to use partitions